src/select/select.js

```diff
@@ -152,21 +154,24 @@ class Select extends React.Component<PropsT, StatelessStateT> {
         break;
       }
       case STATE_CHANGE_TYPE.keyUp: {
-        // $FlowFixMe
-        const newTextValue = e.target.value;
-        this.setState({textValue: newTextValue});
-        if (this.props.filterable) {
-          let filteredOptions = this.props.options.filter(option =>
-            this.props.filterOption(option, newTextValue),
-          );
-          filteredOptions = filteredOptions.length
-            ? filteredOptions
-            : newTextValue
-              ? []
-              : null;
-          this.setState({filteredOptions});
+        if (!this.handledHotKeys(e)) {
+          // $FlowFixMe
+          const newTextValue = e.target.value;
+          this.setState({textValue: newTextValue});
+          if (this.props.filterable) {
+            let filteredOptions = this.props.options.filter(option =>
+              this.props.filterOption(option, newTextValue),
+            );
+            filteredOptions = filteredOptions.length
```

Comment:
maybe it is just me, but I have hard time understanding the double ternary operator here - can we refactor this?

Reply:
ok